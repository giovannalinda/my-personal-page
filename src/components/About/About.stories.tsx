import { Meta, Story } from '@storybook/react'
import { About } from './About'

export default {
  title: 'Components/About',
  component: About,
} as Meta

export const Default: Story = () => <About />
