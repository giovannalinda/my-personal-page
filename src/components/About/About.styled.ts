import { motion } from 'framer-motion'
import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.section`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 32rem;

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    margin-left: 10rem;
  }
`

export const Title = styled(motion.h1)`
  font-size: ${theme.font.sizes.large};
  margin-top: -18rem;
  width: 100%;

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    font-size: ${theme.font.sizes.subtitle};
    margin-top: -38rem;
  }
`

export const Paragraph = styled(motion.p)`
  font-size: ${theme.font.sizes.paragraph};
  color: ${theme.colors.gray400};
  max-width: 48rem;
`
