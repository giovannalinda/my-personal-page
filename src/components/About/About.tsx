import { InView } from 'react-intersection-observer'

import * as S from './About.styled'

const leftVariant = {
  hidden: { opacity: 0, x: -50 },
  visible: { opacity: 1, x: 0, transition: { duration: 0.5 } },
}

const rightVariant = {
  hidden: { opacity: 0, x: 50 },
  visible: { opacity: 1, x: 0, transition: { duration: 0.5 } },
}

export function About() {
  return (
    <InView triggerOnce>
      {({ ref, inView }) => (
        <S.Container ref={ref}>
          {inView && (
            <>
              <S.Title
                variants={leftVariant}
                initial='hidden'
                animate='visible'
              >
                Dedicada a encontrar as melhores soluções criativas para suas
                marcas
              </S.Title>
              <S.Paragraph
                variants={rightVariant}
                initial='hidden'
                animate='visible'
              >
                Me preocupo com cada detalhe na hora de construir interfaces
                valorizando a experiência para que ela impacte positivamente a
                vida dos usuários.
              </S.Paragraph>
            </>
          )}
        </S.Container>
      )}
    </InView>
  )
}
