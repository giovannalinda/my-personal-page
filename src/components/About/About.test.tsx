import { render } from '~/__helpers__/app-tests'
import { About } from './About'

describe('About', () => {
  it('matches snapshot', () => {
    const { container } = render(<About />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
