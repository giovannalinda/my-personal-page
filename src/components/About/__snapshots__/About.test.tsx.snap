// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About matches snapshot 1`] = `
<section
  class="sc-bczRLJ gdEYFF"
>
  <h1
    class="sc-gsnTZi kBOgis"
  >
    Dedicada a encontrar as melhores soluções criativas para suas marcas
  </h1>
  <p
    class="sc-dkzDqf clYJGD"
  >
    Me preocupo com cada detalhe na hora de construir interfaces valorizando a experiência para que ela impacte positivamente a vida dos usuários.
  </p>
</section>
`;
